cdap-ui/app/cdap/components/OpsDashboard/RunsGraph/DataParser.js

```diff
@@ -58,40 +56,32 @@ export function parseDashboardData(rawData, startTime, duration, pipeline, custo
     }
 
     // add status
-    if (endTime && buckets[endTime]) {
+    if (buckets[endTime]) {
       if (runInfo.status === 'COMPLETED') {
         buckets[endTime].successful++;
       } else if (runInfo.status === 'FAILED') {
         buckets[endTime].failed++;
       }
-      buckets[endTime].runsList.push(runInfo);
     }
 
-    // if end time is not present, that means the program is still running
-    let end = runInfo.end * 1000 || Date.now();
-    let start = runInfo.start * 1000;
-
     let startIndex = timeArray.indexOf(startTime);
     // if startTime not found, then the program started before the graph
-    // so set start value to start of first bucket
+    // so set start index to first bucket
     if (startIndex === -1) {
       startIndex = 0;
-      start = parseInt(timeArray[startIndex], 10);
     }
 
     let endIndex = timeArray.indexOf(endTime);
-    // if endTime not found, then the program end after the graph
+    // if endTime not found, then the program ended after the graph
+    // or the program is still running
     if (endIndex === -1) {
-      end = Date.now();
+      endTime = getBucket(Date.now());
+      endIndex = timeArray.indexOf(endTime);
```

Comment:
I think all you need to do here is to get the last index of the timeArray. So `timeArray.length - 1`